src/components/shared/__tests__/Slider.test.tsx

```diff
@@ -1,47 +1,74 @@
-import 'react-native';
-
 import * as React from 'react';
 
+import { fireEvent, render } from '@testing-library/react-native';
+import Rail from '../Slider/Rail';
 import Slider from '../Slider';
-// Note: test renderer must be required after react-native.
-import renderer from 'react-test-renderer';
-
-let props: any;
-let component: React.ReactElement;
-// let testingLib: RenderResult;
-
-const createTestProps = (obj: object): object => ({
-  navigation: {
-    navigate: jest.fn(),
-  },
-  ...obj,
-});
+import { View } from 'react-native';
 
 describe('[Slider] render', () => {
-  beforeEach(() => {
-    props = createTestProps({});
-    component = <Slider {...props} />;
+  it('renders without crashing', () => {
+    const rendered = render(<Slider />).asJSON();
+    expect(rendered).toMatchSnapshot();
+    expect(rendered).toBeTruthy();
   });
+});
+
+describe('[Rail]', () => {
+  const TEST_ID = 'TEST_ID_FOR_MARK';
+  const Mark: React.FC<{ testID: string }> = ({ testID }) => {
+    return <View testID={testID} />;
+  };
 
   it('renders without crashing', () => {
-    const rendered: renderer.ReactTestRendererJSON | null = renderer
-      .create(component)
-      .toJSON();
+    const rendered = render(
+      <Rail mark={<Mark testID={TEST_ID} />} customMarkWidth={1} />,
+    ).asJSON();
     expect(rendered).toMatchSnapshot();
     expect(rendered).toBeTruthy();
   });
 
-  // describe('interactions', () => {
-  //   beforeEach(() => {
-  //     testingLib = render(component);
-  //   });
-
-  //   it('should simulate onClick', () => {
-  //     const btn = testingLib.queryByTestId('btn');
-  //     act(() => {
-  //       fireEvent.press(btn);
-  //     });
-  //     expect(cnt).toBe(3);
-  //   });
-  // });
+  it('calls onMarkPress with correct value and index', () => {
+    const onMarkPress = jest.fn();
+    const STEP = 10;
+    const MARK_COUNT = 5;
+
+    const { getAllByTestId } = render(
+      <Rail
+        mark={<Mark testID={TEST_ID} />}
+        customMarkWidth={10}
+        step={STEP}
+        markCount={MARK_COUNT}
+        onMarkPress={onMarkPress}
+      />,
+    );
+    const marks = getAllByTestId(TEST_ID);
+    for (let i = 0; i < MARK_COUNT; i += 1) {
+      fireEvent.press(marks[i]);
+    }
+    for (let i = 0; i < MARK_COUNT; i += 1) {
+      expect(onMarkPress.mock.calls[i][0]).toBe(i * STEP);
+      expect(onMarkPress.mock.calls[i][2]).toBe(i);
+    }
+  });
+
+  it('calls onInit', () => {
+    const onInit = jest.fn();
+    const STEP = 8;
+    const MARK_COUNT = 4;
+
+    render(
+      <Rail
+        mark={<Mark testID={TEST_ID} />}
+        customMarkWidth={12}
+        style={{ width: 240 }}
+        step={STEP}
+        markCount={MARK_COUNT}
+        onInit={onInit}
+      />,
+    );
+    expect(onInit).toHaveBeenCalledTimes(1);
+    expect(onInit.mock.calls[0][0]).toStrictEqual(
```

Comment:
How about using `destructuring` syntax or other ways like extracting getter for `mock.calls` to recognize the context intuitively?
Tests should explain in business words not technical, to understand what it means easily and fast.
That reduces the effort of reading and inference to maintain both `real` and `test`.

Reply:
I have changed the code to convey the meaning of this test in business words

```
const expectedArgumentForOnInit = Array.from({ length: MARK_COUNT }).map((_, ind) => STEP * ind);
const actualArgumentForOnInit = onInit.mock.calls[0][0];
expect(actualArgumentForOnInit).toStrictEqual(expectedArgumentForOnInit);
```